test/unified_memory_allocation/uma_pools/disjoint_pool_config.cpp

```diff
@@ -0,0 +1,233 @@
+                   {SHARED, PoolDescriptor(UR_USM_TYPE_SHARED)},
+                   {READ_ONLY_SHARED, PoolDescriptor(UR_USM_TYPE_SHARED, true)}};
+
+class disjointPoolConfigTests : public testing::TestWithParam<std::string> {
+  protected:
+    const int hostMaxPoolableSizeDefault = 2 * 1024 * 1024;
+    const int hostCapacityDefault = 4;
+    const int hostSlabMinSizeDefault = 64 * 1024;
+
+    const int deviceMaxPoolableSizeDefault = 4 * 1024 * 1024;
+    const int deviceCapacityDefault = 4;
+    const int deviceSlabMinSizeDefault = 64 * 1024;
+
+    const int sharedMaxPoolableSizeDefault = 0;
+    const int sharedCapacityDefault = 0;
+    const int sharedSlabMinSizeDefault = 2 * 1024 * 1024;
+
+    const int readOnlySharedMaxPoolableSizeDefault = 4 * 1024 * 1024;
+    const int readOnlySharedCapacityDefault = 4;
+    const int readOnlySharedSlabMinSizeDefault = 2 * 1024 * 1024;
+
+    const int maxSizeDefault = 16 * 1024 * 1024;
+};
+
+// invalid configs for which descriptors' parameters should be set to default values
+INSTANTIATE_TEST_SUITE_P(configsForDefaultValues,
+                         disjointPoolConfigTests,
+                         testing::Values("",
+                                         "ab12cdefghi34jk56lmn78opr910",
+                                         "1;32M;foo:0,3,2m;device:1M,4,64k",
+                                         "132M;host:4m,3,2m,4m;device:1M,4,64k",
+                                         "1;32M;abdc123;;;device:1M,4,64k",
+                                         "1;32M;host:0,3,2m,4;device:1M,4,64k,5;host:1,8,4m,100",
+                                         "32M;1;host:1M,4,64k;device:1m,4,64K;shared:0,3,1M"));
+
+TEST_F(disjointPoolConfigTests, disjointPoolConfigTest) {
+    // test if the returned unordered map contains only the valid configs
+    usm_pool::PoolConfigurations<typename DisjointPool::Config> configsMap =
+        usm_pool::getPoolConfigurationsFor<DisjointPool>();
+
+    // check if all the valid configs are present
+    for (const auto &descriptor : DESCRIPTORS) {
+        ASSERT_EQ(configsMap.count(descriptor.second), 1);
+    }
+
+    // check if there are no more configs in the unordered map
+    ASSERT_EQ(configsMap.size(), 4);
+}
+
+TEST_F(disjointPoolConfigTests, disjointPoolConfigStringEnabledBuffersTest) {
+    // test for valid string with enabled buffers-- (values to configs should be
+    // parsed from string)
+    std::string config = "1;32M;host:1M,4,64k;device:1m,4,64K;shared:0,3,1M;"
+                         "read_only_shared:0,0,3M";
+    usm_pool::PoolConfigurations<typename DisjointPool::Config> configsMap =
+        usm_pool::getPoolConfigurationsFor<DisjointPool>(config);
+
+    // test for host
+    ASSERT_EQ(configsMap.at(DESCRIPTORS.at(HOST)).MaxPoolableSize, 1 * 1024 * 1024);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS.at(HOST)).Capacity, 4);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS.at(HOST)).SlabMinSize, 64 * 1024);
```

Comment:
Check for values different than defaults - currently you can't tell if it's set to config string's value or the default one, because they are equal.

Reply:
Done.